attractions/autocomplete/autocomplete-field.svelte

```diff
@@ -13,13 +19,49 @@
   let _class = null;
   export { _class as class };
 
+  /**
+   * An async generator of suggestions.
+   * Receives the input from the text field and is expected to yield promises that resolve to arrays of objects
+   * @type {OptionsGetter}
+   */
   export let getOptions;
+  /**
+   * The current selection as an array of objects.
+   * Can be used to set the selection programmatically.
+   * @type {Option[]}
+   */
   export let selection = [];
+  /**
+   * The minimum amount of characters to call `getOptions`
+   * @type {number}
+   */
   export let minSearchLength = 3;
+  /**
+   * The maximum amount of options than can be selected.
+   * @type {number}
+   */
   export let maxOptions = Infinity;
+  /**
+   * The current value of the text field. Can be used to control the query programmatically.
+   * @type {string}
+   */
   export let searchQuery = '';
+  /**
+   * The component used to render a suggestion in the list.
+   * Receives two props: the option object ({ name, details }) and the current query.
+   * Expected to dispatch click events on selection
+   * @type {SvelteComponentTyped<{option: Option; query: string}, {click: MouseEvent}, {}>}
```

Comment:
Not exactly a `MouseEvent`, it can also be a custom event with any details, so maybe it's worth relaxing the condition here

Reply:
You mean make it just `Event` or `any` or what exactly?

Reply:
We can make it a `MouseEvent | CustomEvent` with any template argument.

Reply:
Well, it's not like we are using this event anyways, so I just made it `Event` since both of these inherit from it.